src/neuroconv/tools/text.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, Optional
+
+import numpy as np
+import pandas as pd
+from pynwb.epoch import TimeIntervals
+
+
+def convert_df_to_time_intervals(
+    df: pd.DataFrame,
+    name: str = "trials",
+    description: str = "experimental trials",
+    column_name_mapping: Dict[str, str] = None,
+    column_descriptions: Dict[str, str] = None,
+) -> TimeIntervals:
+    """
+    Convert a dataframe to a TimeIntervals object.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        The dataframe to convert.
+    name : str, optional
+        The name of the TimeIntervals object.
+    description : str, optional
+        The description of the TimeIntervals object.
+    column_name_mapping: dict, optional
+        If passed, rename subset of columns from key to value.
+    column_descriptions: dict, optional
+        Keys are the names of the columns (after renaming) and values are the descriptions. If not passed,
+        the names of the columns are used as descriptions.
+
+    Returns
+    -------
+    TimeIntervals
+
+    """
+    if column_name_mapping is not None:
+        df.rename(columns=column_name_mapping, inplace=True)
+
+    default_column_descriptions = dict(
+        start_time="Start time of epoch, in seconds.",
+        stop_time="Stop time of epoch, in seconds.",
+    )
+
+    if column_descriptions is None:
+        column_descriptions = default_column_descriptions
+    else:
+        column_descriptions = dict(default_column_descriptions, **column_descriptions)
+
+    if description is None:
+        description = name
```

Comment:
Feel like this is duplication of information - default table description could at the very least be some bit of extra information, like `"Autogenerated by NeuroConv."`. The interval usage could also make use of the `file_path` by writing something like `"This TimeIntervals table was converted from the source file {file_path}" using NeuroConv.` which leaves additional provenance of how that table came to be (either experimentally, or by way of NWB conversion)

Reply:
I like the idea of including the source filepath in the description